qiskit/aqua/components/optimizers/gsls.py

```diff
@@ -0,0 +1,325 @@
+                'step_size_multiplier', 'armijo_parameter',
+                'min_gradient_norm', 'max_failed_rejection_sampling']
+
+    # pylint:disable=unused-argument
+    def __init__(self,
+                 max_iter: int = 10000,
+                 max_eval: int = 10000,
+                 disp: bool = False,
+                 sampling_radius: float = 1.0e-6,
+                 sample_size_factor: int = 1,
+                 initial_step_size: float = 1.0e-2,
+                 min_step_size: float = 1.0e-10,
+                 step_size_multiplier: float = 0.4,
+                 armijo_parameter: float = 1.0e-1,
+                 min_gradient_norm: float = 1e-8,
+                 max_failed_rejection_sampling: int = 50) -> None:
+        """
+        Args:
+            max_iter: Maximum number of iterations.
+            max_eval: Maximum number of evaluations.
+            disp: Set to True to display convergence messages.
+            sampling_radius: Sampling radius to determine gradient estimate.
+            sample_size_factor: The size of the sample set at each iteration is this number
+                multiplied by the dimension of the problem, rounded to the nearest integer.
+            initial_step_size: Initial step size for the descent algorithm.
+            min_step_size: Minimum step size for the descent algorithm.
+            step_size_multiplier: Step size reduction after unsuccessful steps, in the
+                interval (0, 1).
+            armijo_parameter: Armijo parameter for sufficient decrease criterion, in the
+                interval (0, 1).
+            min_gradient_norm: If the gradient norm is below this threshold, the algorithm stops.
+            max_failed_rejection_sampling: Maximum number of attempts to sample points within
+                bounds.
+        """
+        super().__init__()
+        for k, v in locals().items():
+            if k in self._OPTIONS:
+                self._options[k] = v
+
+    def get_support_level(self) -> Dict[str, int]:
+        """Return support level dictionary.
+
+        Returns:
+            A dictionary containing the support levels for different options.
+        """
+        return {
+            'gradient': Optimizer.SupportLevel.ignored,
+            'bounds': Optimizer.SupportLevel.supported,
+            'initial_point': Optimizer.SupportLevel.required
+        }
+
+    def optimize(self, num_vars: int,
+                 objective_function: callable,
+                 gradient_function: Optional[callable] = None,
+                 variable_bounds: Optional[List[Tuple[float, float]]] = None,
+                 initial_point: Optional[np.ndarray] = None) -> Tuple[np.ndarray, float, int]:
+        super().optimize(num_vars, objective_function, gradient_function,
+                         variable_bounds, initial_point)
+
+        if initial_point is None:
+            initial_point = np.random.normal(size=num_vars)
```

Comment:
Can we have this use aqua globals defined random rather  than directly using numpy random. This was we get control of tests etc through just setting aqua seed and do not have to go directly to numpy. I see more than one location in this file.